mlflow/tracking/llm_utils.py

```diff
@@ -0,0 +1,90 @@
+import logging
+from typing import Dict, List, Union
+
+_logger = logging.getLogger(__name__)
+
+
+def log_predictions(
+    inputs: List[Union[str, Dict[str, str]]],
+    outputs: List[str],
+    prompts: List[Union[str, Dict[str, str]]],
+) -> None:
+    """
+    Log a batch of inputs, outputs and prompts for the current evaluation run.
+    If no run is active, this method will create a new active run.
+
+    :param inputs: Union of either List of input strings or List of input dictionary
+    :param outputs: List of output strings
+    :param prompts: Union of either List of prompt strings or List of prompt dictionary
+    :returns: None
+
+    .. test-code-block:: python
+        :caption: Example
+
+        import mlflow
+
+        inputs = [{
+            "question": "How do I create a Databricks cluster with
+                        UC access?",
+            "context": "Databricks clusters are ..."
+        },
+        ...
+        ]
+
+        outputs = [
+            "<Instructions for cluster creation with UC enabled>",
+        ]
+
+        prompts = [
+            "Get Databricks documentation to answer all the questions: {input}",
+        ]
+
+
+        # Log llm predictions
+        with mlflow.start_run():
+            mlflow.llm.log_predictions(inputs, outputs, prompts)
+    """
+    if len(inputs) != len(outputs) or len(inputs) != len(prompts):
+        raise ValueError("The length of inputs, outputs and prompts must be the same.")
+
+    import csv
+    import mlflow
+
+    header = []
+    data = []
+    run_id = mlflow.tracking.fluent._get_or_start_run().info.run_id
+    LLM_ARTIFACT_NAME = "llm_predictions.csv"
+
+    try:
+        artifact_path = mlflow.artifacts.download_artifacts(
+            run_id=run_id, artifact_path=LLM_ARTIFACT_NAME
+        )
+        import csv
+
+        with open(artifact_path, newline="") as llm_artifact:
+            data = list(csv.reader(llm_artifact))
```

Comment:
Can we just append the new items instead of rewriting the entire content?

Reply:
Interesting. Let me test if we can append to an existing file with the `csv` module.